geps/gep-1651.md

```diff
@@ -0,0 +1,408 @@
+
+### Scope/reachability/routability field on GatewayAddress
+
+This would allow Gateways to have multiple scopes.
+
+From: https://github.com/kubernetes-sigs/gateway-api/pull/1653#issuecomment-1486271913
+> The obvious application for multiple scopes seems to be saving on boilerplate, which is a win, but are there are any other advantages to allowing one Gateway to have multiple scopes?
+>
+> Multiple scopes Pros:
+>
+> Allows a single Gateway to express multiple networks, saving on needing to attach HTTPRoutes to multiple Gateways for each network scope.
+>
+> Multiple scopes Cons:
+>
+> Complicates the Gateway's purpose. Instead of one Gateway being one set of Listeners, now a Gateway is two sets of listeners that have a totally different scope (and presumably, security context). Personally, I'm also concerned how this will interact with other features like merging and preprovisioning that GEP-1867: Per-Gateway Infrastructure #1868 will allow.
+
+### Adding `routability` attribute to GatewayClass
+
+See [Prior Art - Multiple Gateways Classes](#multiple-gateway-classes)
+
+## Survey of Prior Art
+
+These alternatives are a survey of existing approaches to support cluster
+local Gateways. Most are implementation specific and are not portable.
+
+### Special annotation/label
+
+Istio let's you specify an annotation `networking.istio.io/service-type` to
+change the underlying Kubernetes Service type to make it a ClusterIP type.
+
+### Re-use of AddressType Hostname
+
+Istio let's you re-use existing Gateway deployments by setting the address
+type to `Hostname` and the value to the Istio ingress Kubernetes Service. If an
+operator configures the Istio deployment to support cluster local traffic a
+Gateway implementation can select it using the `HostName` attribute.
+
+### Multiple Gateway Classes
+
+Some implementations support multiple deployments on a single cluster where each maps to a
+GatewayClass. One of these deployments can be configured to serve cluster local traffic. This is
+sub-optimal because this is implementation specific and the end-user is effectively managing the
+deployments themselves rather than infrastructure being automatically provisioned.
+
+Likewise, infrastructure providers may provide a fixed set of GatewayClasses with unique and fixed
+routability. Thus GatewayClass name is a viable option to control routability. There may be a
+non-zero cost when requiring additional GatewayClasses - but this depends on the implementation.
+
+Additionally, if more attributes are added to GatewayClass to constrain Gateways in some
+form this leads to a combinatorial number of GatewayClassNames. For example, `foo-public` and
+`foo-cluster` are two GatewayClasses surfacing the values of a single attribute `routability`.
+Let's say we want to enforce address types to just IP then our `gatewayClassName` would be:
+- `foo-public-ipv4`
```

Comment:
Need line break for list formatting: https://deploy-preview-1653--kubernetes-sigs-gateway-api.netlify.app/geps/gep-1651/#multiple-gateway-classes

```suggestion
Let's say we want to enforce address types to just IP then our `gatewayClassName` would be:

- `foo-public-ipv4`
```